backend/src/data/repositories/course/course.repository.ts

```diff
@@ -23,6 +23,10 @@ class Course {
       courseCategoryId,
     });
   }
+
+  async findByName(name: string): Promise<CourseM[]> {
+    return this.#CourseModel.query().where('title', 'like', `%${name}%`);
```

Comment:
```suggestion
 findByName(name: string): Promise<CourseM[]> {
    return this.#CourseModel.query().where('title', 'like', `%${name}%`).execute();
```
more info [https://github.com/BinaryStudioAcademy/bsa-2022-guruhub/pull/103#discussion_r945107162](https://github.com/BinaryStudioAcademy/bsa-2022-guruhub/pull/103#discussion_r945107162)